### ISSUE TITLE
Support creating temporary directory that is recursively deleted on drop

### ISSUE THREAD

**Issue by ibotty** [sunsided/async-tempfile-rs#5]
My use case is I'll need a directory where I'll put some files.  Then some other process will generate files within that directory and I'll upload the generated files.

**Comment by sunsided** [comment]
@ibotty I drafted some work for that in #8. Could you try using the crate from source and tell me if it works for your case?

```toml
async-tempfile = { git = "https://github.com/sunsided/async-tempfile-rs.git", branch = "feature/tempdir" }
```

**Comment by ibotty** [comment]
Great!  Thank you.  I'll have a look in the next week.

**Comment by ibotty** [comment]
Sorry for not testing earlier.  It works like a charm!  I replaced my (non-async) `tempfile`'s `TempDir` usage and only had to add a few `await`s and change `path` to `dir_path`.  Thank you!

**Comment by sunsided** [comment]
@ibotty Thank you! It's available in [0.6.0](https://github.com/sunsided/async-tempfile-rs/releases/tag/v0.6.0) now. ✌🏼
